### DISCUSSION TITLE
More than 8 encoders

### DISCUSSION THREAD

**Discussion by marcoscandella** [tttapa/Control-Surface#683] [Q&A]
Hi, thanks in advance for your reply.
I need to create a Midi controller with 20 rotary encoders and 12 buttons (with Arduino Due)
With this library I managed to put only 8 .. can you help me? could you dedicate the CCs used for all the other commands?

**Comment by benwadub** [comment]
hi, maybe you could think about moving to a teensy? I use 24 encoders with push buttons and 68 other buttons and 124 potentiometer without problem on only one teensy 4.1 look at the mcp23017 class for the encoders

**Comment by tttapa** [comment]
Control Surface doesn't care what addresses you use for the encoders, you can use controllers 0 through 119 (and even 120-127 if your DAW supports it) on all 16 MIDI channels for a total of 1920 controls. You don't have to use the eight `MCU::V_POT` constants, you can just specify a number.
